blueprints/layered-profile.md

```diff
@@ -0,0 +1,347 @@
+
+**Existing Endpoints**
+
+- Modify JSON request and response for existing servers endpoints.
+
+#### API 5.0 GET
+- JSON **response** with the proposed change will look as follows:
+`/servers?id=5`
+```JSON
+{ 
+  "id": 5,
+  ⋮
+  "requested": {
+    "layeredProfiles": [
+      "MID"
+    ],
+    ⋮
+  }
+}
+```
+
+`/servers`
+```JSON
+{ 
+  "requested": {
+    "layeredProfiles": [
+      "MID"
+    ],
+    ⋮
+  },
+  ⋮
+}
+```
+
+#### API 5.0 POST/PUT
+JSON **request** with the proposed change will look as follows:
+
+`POST /servers `
+```JSON
+{
+    "cachegroupId": 6,
+    "cdnId": 2,
+    ⋮
+    "interfaces": [
+      ⋮
+    ],
+    ⋮
+    "profileId": [10, 11, 15],
+    ⋮
+}
+```
+
+`PUT /servers/5`
+```JSON
+{
+    "cachegroupId": 6,
+    "cdnId": 2,
+    ⋮
+    "interfaces": [
+      ⋮
+    ],
+    ⋮
+    "profileId": [10, 11, 15],
+    ⋮
+}
+```
+
+returns a **400** **response** with
+
+```JSON
+{
+  "alerts":[
+   { 
+     "text":"cannot associate multiple profiles to a server with current API version but can be done via 5.0+",
+     "level":"error"
+   }
+]}
+```
+
+#### API 6.0 GET
+- JSON **response** with the proposed change will look as follows:
+  `/servers?id=5`
+```JSON
+{ 
+  "id": 5,
+  ⋮
+  "requested": {
+    "layeredProfiles": [
+      "MID",
+      "AMIGA_123",
+      "CDN_FOO"
+    ],
+    ⋮
+  }
+}
+```
+
+`/servers`
+```JSON
+{ 
+  "requested": {
+    "layeredProfiles": [
+      "MID",
+      "AMIGA_123",
+      "CDN_FOO"
+    ],
+    ⋮
+  },
+  ⋮
+}
+```
+
+#### API 6.0 POST/PUT
+JSON **request** with the proposed change will look as follows:
+
+`POST /servers `
+```JSON
+{
+    "cachegroupId": 6,
+    "cdnId": 2,
+    ⋮
+    "interfaces": [
+      ⋮
+    ],
+    ⋮
+    "profileId": [10, 11, 15],
+    ⋮
+}
+```
+
+`PUT /servers/5`
+```JSON
+{
+    "cachegroupId": 6,
+    "cdnId": 2,
+    ⋮
+    "interfaces": [
+      ⋮
+    ],
+    ⋮
+    "profileId": [10, 11, 15],
+    ⋮
+}
+```
+
+return following **response**
+```JSON
+{ "alerts": [
+  {
+    "text": "Server created/updated",
+    "level": "success"
+  }
+],
+  "response": {
+    "cachegroup": "CDN_in_a_Box_Mid",
+    "cachegroupId": 6,
+    "cdnId": 2,
+    ⋮
+    "id": 5,
+    "layeredProfiles": [
+      "MID",
+      "AMIGA_123",
+      "CDN_FOO"
+    ],
+    "profileDesc": "Mid Cache - Apache Traffic Server, Amiga 123 Machine, CDN Foo",
+    "profileIds": [10, 11, 15],
+    ⋮
+  }
+}
+```
+
+The following table describes the top level `layered_profile` object for servers:
+
+| field           | type                 | optionality | description                                              |
+| ----------------| ---------------------| ----------- | ---------------------------------------------------------|
+| server          | bigint               | required    | the server id associated with a given profile            |
+| layeredProfiles | text                 | required    | the profile names associated with a server               |
```

Comment:
This is really just the name of a single Profile, right? That's why it has `order`, and why we need a table at all, I thought.

Reply:
You are right, i'll change it the description.